server/routes/changeGoal/ChangeGoalController.ts

```diff
@@ -61,6 +61,11 @@ export default class ChangeGoalController {
       const planUuid = req.services.sessionService.getPlanUUID()
       const plan = await req.services.planService.getPlanByUuid(planUuid)
 
+      // Check if the user has came back from the add-goals page
+      if (req.services.sessionService.getReturnLink() === `/change-goal/${goalUuid}/`) {
+        req.services.sessionService.setReturnLink(null)
+        redirectTarget = `/goal/${goalUuid}/add-steps`
```

Comment:
Do we not want to keep something set in the returnLink variable? Could this maybe cause a null exception somewhere? or a user being redirected to `null`?

Reply:
Yep, you're right. I can't actually remember why I added this line in. I'll remove it so it remains set to the current returnLink (which solves a problem on the next page in the journey)